### ISSUE TITLE
Add support for samp and kbd in pl-code

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#10594]
The `samp` and `kbd` semantic tags can be used to provide [examples of terminal output and input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/samp#sample_output_including_user_input). It is currently possible to add them to a `<pre>` block, but not to `<pl-code>`, thus losing on features like styles and line highlighting.

For example, I can use:

```html
<pre><samp>Enter the distance travelled (in meters): <kbd>100</kbd>
Enter the time taken (in seconds): <kbd>2.5</kbd>
40.0
</samp></pre>
```

and generate:

<pre><samp>Enter the distance travelled (in meters): <kbd>100</kbd>
Enter the time taken (in seconds): <kbd>2.5</kbd>
40.0
</samp></pre>

**Comment by nwalters512** [comment]
I'm struggling to understand the use case here. When would you want to intermingle these in a code block? I certainly can't do that in a text editor. Is the fact that you can't use what are _technically_ the correct semantic tags an accessibility issue? VoiceOver does absolutely nothing useful with your example above, it's all just read as text.

**Comment by jonatanschroeder** [comment]
Maybe I should rephrase this to clarify the practical use case I'm hoping to use: I'd like to have a way to demonstrate to students the input/output expected of a particular program. This process should have a way to visually distinguish between what is output (printed by the program) and what is input (typed by the user). The use of `samp` and `kbd` inside a `pre` block is the HTML-sanctioned way to do this, but the current styles used in PL/bootstrap don't make this markup very clear, and don't make a good visual distinction (e.g., border, background) between this terminal output block and the rest of the text.

In the past I used pl-code to represent this kind of output, with `language="console"`, however pl-code does not give support to the distinction between input and output. Maybe my suggestion above of adding this distinction to pl-code may not be a good idea, and instead a different approach could be used.

For now I'm using my own styling and using specific classes in the `pre` block, which may be a reasonable solution in this context, but it would be good to have a way to have this styling be closer to what pl-code itself uses. If you believe the best approach here is to use course-specific styles, I understand, and I'm happy to adapt, but having something that is a bit more PL-flavoured for this kind of use case would probably be visually better. I won't be offended if you believe this is my own problem and decide to close the issue.

**Comment by nwalters512** [comment]
This seems outside the scope of what Pygments, and by extension `<pl-code>`, is meant to do. I see no evidence that we can configure it to selectively treat portions of the code we give it as HTML instead of code to run through the lexer, let alone to have everything downstream of the lexer work as you're describing.

I view this as largely similar to the distinction between `<pre>` and code fences in GitHub's editors like you used in the original issue. There's no way to use `<kbd>` / `<samp>` in a code fence, but you can use them when you drop down into raw HTML via `<pre>`.
